lib/livebook/ecto_types/secret_origin.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Livebook.EctoTypes.SecretOrigin do
+  @moduledoc false
+  use Ecto.Type
+
+  @type t :: :session | :startup | :app | {:hub, String.t()}
+
+  def type, do: :any
+
+  def load("session"), do: {:ok, :session}
+  def load("app"), do: {:ok, :app}
+  def load("startup"), do: {:ok, :startup}
+
+  def load(id) when is_binary(id) do
+    if hub_secret?(id),
+      do: {:ok, {:hub, id}},
+      else: :error
+  end
+
+  def load(_), do: :error
+
+  def dump(:session), do: {:ok, "session"}
+  def dump(:app), do: {:ok, "app"}
+  def dump(:startup), do: {:ok, "startup"}
+  def dump({:hub, id}) when is_binary(id), do: dump(id)
+
+  def dump(id) when is_binary(id) do
+    if hub_secret?(id), do: {:ok, id}, else: :error
+  end
```

Comment:
```suggestion

  def dump({:hub, id}) when is_binary(id) do
    if hub_secret?(id) do
      {:ok, "hub-" <> id}
    else
      :error
    end
  end
```

Reply:
But it would be `hub-enterprise-[long id]`. Is it the expected behaviour?

Reply:
Here we assume the id to be anything, so it sounds fine to me. It's just that if we ever have another source with id we need to distinguish, so it's better to tag every origin.

Reply:
I mean, every hub has a prefix which means where it comes from `fly-{huge id}`, `enterprise-{huge id}`, etc.

Reply:
But it's more like `hub_id = fly-{fly_id}`, right? For `:origin` the hub id is opaque, we just need to know it's a hub, and we should be able to tell without checking if a hub with that id exists.